internal/lib/lib.go

```diff
@@ -670,9 +671,16 @@ func VrfChecksum(vrfName string, staticRoutes []*models.StaticRoute) uint32 {
 	return utils.Hash(utils.Stringify(filteredStaticRoutes))
 }
 
-func DSChecksum(pgrefs []string) uint32 {
+func DSChecksum(pgrefs []string, markers []*models.RoleFilterMatchLabel) uint32 {
 	sort.Strings(pgrefs)
 	checksum := utils.Hash(utils.Stringify(pgrefs))
+	if GetGRBACSupport() {
+		if markers != nil {
+			checksum += ObjectLabelChecksum(markers)
+		} else {
+			checksum += GetClusterLabelChecksum()
```

Comment:
@chauhanshubham : This will require same logic as that of ObjectLabelChecksum(markers) function.  If we consider existing 20.1.4 deployments where existing DS, L4 etc. objects will not labeled. So here if we add label cksum by default to it while adding it to cache , we will not do put call for adding labels at layer2 as computed and cache cksum will be same and those objects will not be labelled at controller side.

Reply:
Yeah I think I am not able to understand why we have to ever add the ClusterLabelChecksum? the GRBAC support is either with markers or it does not exist.

Reply:
I think.. he is trying to support label stuff also.. as it is there in 20.1.4.

Reply:
This does not change the existing functionality, just moves the checksum calculation based on labels inside the helper function. This might have been incorrect before as well?
cc: @parimanur